src/sinks/gcp/stackdriver/logs/tests.rs

```diff
@@ -0,0 +1,314 @@
+//! Unit tests for the `gcp_stackdriver_logs` sink.
+
+use bytes::Bytes;
+use chrono::{TimeZone, Utc};
+use futures::{future::ready, stream};
+use http::{Request, Uri};
+use indoc::indoc;
+use lookup::lookup_v2::ConfigValuePath;
+use serde::Deserialize;
+use serde_json::value::RawValue;
+use std::collections::HashMap;
+
+use crate::{
+    config::{GenerateConfig, SinkConfig, SinkContext},
+    event::{LogEvent, Value},
+    gcp::GcpAuthenticator,
+    sinks::{
+        gcp::stackdriver::logs::{config::StackdriverLogName, encoder::remap_severity},
+        prelude::*,
+        util::BoxedRawValue,
+    },
+    test_util::{
+        components::{run_and_assert_sink_compliance, SINK_TAGS},
+        http::{always_200_response, spawn_blackhole_http_server},
+    },
+};
+
+use super::{
+    config::{default_endpoint, StackdriverConfig, StackdriverResource},
+    encoder::StackdriverLogsEncoder,
+};
+
+#[test]
+fn generate_config() {
+    crate::test_util::test_generate_config::<StackdriverConfig>();
+}
+
+#[tokio::test]
+async fn component_spec_compliance() {
+    let mock_endpoint = spawn_blackhole_http_server(always_200_response).await;
+
+    let config = StackdriverConfig::generate_config().to_string();
+    let mut config = StackdriverConfig::deserialize(toml::de::ValueDeserializer::new(&config))
+        .expect("config should be valid");
+
+    // If we don't override the credentials path/API key, it tries to directly call out to the Google Instance
+    // Metadata API, which we clearly don't have in unit tests. :)
+    config.auth.credentials_path = None;
+    config.auth.api_key = Some("fake".to_string().into());
+    config.endpoint = mock_endpoint.to_string();
+
+    let context = SinkContext::default();
+    let (sink, _healthcheck) = config.build(context).await.unwrap();
+
+    let event = Event::Log(LogEvent::from("simple message"));
+    run_and_assert_sink_compliance(sink, stream::once(ready(event)), &SINK_TAGS).await;
```

Comment:
Are we able to change this to `HTTP_SINK_TAGS`?
```suggestion
    run_and_assert_sink_compliance(sink, stream::once(ready(event)), &HTTP_SINK_TAGS).await;
```